File: app/src/main/java/eu/kanade/tachiyomi/widget/MinMaxNumberPicker.java
package eu.kanade.tachiyomi.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.widget.NumberPicker;

import eu.kanade.tachiyomi.R;

public class MinMaxNumberPicker extends NumberPicker{

    public MinMaxNumberPicker(Context context) {
        super(context);
    }

    public MinMaxNumberPicker(Context context, AttributeSet attrs) {
        super(context, attrs);
        processAttributeSet(context, attrs);
    }

    public MinMaxNumberPicker(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        processAttributeSet(context, attrs);
    }

    private void processAttributeSet(Context context, AttributeSet attrs) {
        TypedArray ta = context.obtainStyledAttributes(attrs, R.styleable.MinMaxNumberPicker, 0, 0);
        try {
            setMinValue(ta.getInt(R.styleable.MinMaxNumberPicker_min, 0));
            setMaxValue(ta.getInt(R.styleable.MinMaxNumberPicker_max, 0));
        } finally {
            ta.recycle();
        }
    }
}

File: app/src/main/java/eu/kanade/tachiyomi/widget/MinMaxNumberPicker.kt
package eu.kanade.tachiyomi.widget

import android.content.Context
import android.text.InputType
import android.util.AttributeSet
import android.widget.EditText
import android.widget.NumberPicker
import androidx.core.view.doOnLayout
import eu.kanade.tachiyomi.R
import eu.kanade.tachiyomi.util.view.findDescendant

class MinMaxNumberPicker @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) :
    NumberPicker(context, attrs) {

    override fun setDisplayedValues(displayedValues: Array<out String>?) {
        super.setDisplayedValues(displayedValues)

        // Disable keyboard input when a value that can't be auto-filled with number exists
        val notNumberValue = displayedValues?.find { it.getOrNull(0)?.digitToIntOrNull() == null }
        if (notNumberValue != null) {
            descendantFocusability = FOCUS_BLOCK_DESCENDANTS
        }
    }

    init {
        if (attrs != null) {
            val ta = context.obtainStyledAttributes(attrs, R.styleable.MinMaxNumberPicker, 0, 0)
            try {
                minValue = ta.getInt(R.styleable.MinMaxNumberPicker_min, 0)
                maxValue = ta.getInt(R.styleable.MinMaxNumberPicker_max, 0)
            } finally {
                ta.recycle()
            }
        }

        doOnLayout {
            findDescendant<EditText>()?.setRawInputType(InputType.TYPE_CLASS_NUMBER)
        }
    }
}
